test/models/patient_test.rb

```diff
@@ -78,6 +81,31 @@ class PatientTest < ActiveSupport::TestCase
     end
   end
 
+  describe 'pledge_summary' do
+    it "should return sent and remaining pledge for the next 7 days" do
+      # Doesn't error when there are no pregnancies
+      summary = Patient.pledged_status_summary
+      assert_equal '{:pledged=>0, :sent=>0}', summary.to_s
+      [@patient, @patient2].each do |pt|
+        create :pregnancy, patient: pt, created_by: @user
+      end
+      @patient.update appointment_date: ( Date.today + 4 )
+      @patient.pregnancy.update dcaf_soft_pledge: 300
+      @patient2.update appointment_date: ( Date.today + 8 )
+      @patient2.pregnancy.update dcaf_soft_pledge: 500
```

Comment:
You can one-liner these two transformations to the pregnancy! `...pregnancy.update dcaf_soft_pledge: 500, pledge_sent: true` will do 'em both in one fell swoop. Update takes a hash and sets stuff with those keys to the values. You might be able to also one-liner the appointment date as `patient.update appointment_date: (Date.today + 4), pregnancy: { dcaf_soft_pledge: ... }` but I forget whether that's something that works or whether that's something that I think should work and doesn't 💯